esiclient/v1/orchestrator/openshift.py

```diff
@@ -0,0 +1,339 @@
+        response = requests.get(full_url, headers=headers)
+    elif method == 'patch':
+        response = requests.patch(full_url, headers=headers, json=data)
+    else:
+        raise OSAIException(
+            'Unknown method used when calling Assisted Installer')
+    if response.status_code not in [200, 201, 202, 204]:
+        raise OSAIException(
+            "Unexpected response from Assisted Installer (%s): %s"
+            % (response.status_code, response.reason))
+    return response
+
+
+def wait_for_nodes(infra_env_id, headers, num_nodes, target_status):
+    waiting = True
+    print("waiting for hosts to reach %s..." % target_status)
+    while waiting:
+        response = call_assisted_installer_api(
+            "infra-envs/%s/hosts" % infra_env_id,
+            'get', headers)
+        count = 0
+        print("* checking hosts")
+        for host in response.json():
+            status = host.get('status')
+            print("  * " + host.get('requested_hostname') + ": " + status)
+            if status == target_status:
+                count = count + 1
+        if count == num_nodes:
+            waiting = False
+        else:
+            time.sleep(30)
+    print("... hosts ready")
+    return
+
+
+class OSAIException(Exception):
+    pass
+
+
+class Orchestrate(command.Lister):
+    """Orchestrate an OpenShift cluster using the Assisted Installer API"""
+
+    log = logging.getLogger(__name__ + ".Orchestrate")
+
+    def _print_failure_message(self, cluster_config_file, cluster_id=None,
+                               infra_env_id=None, message=None):
+        flags = ""
+        if cluster_id:
+            flags = "--cluster-id %s" % cluster_id
+            if infra_env_id:
+                flags = "%s --infra-env-id %s" % (flags, infra_env_id)
+        command = "openstack esi orchestrate openshift %s %s" % \
+            (flags, cluster_config_file)
+        if message:
+            print(message)
+        print("You may need to refresh your OpenShift token.")
+        print("Run this command to continue installation: %s" % command)
+        return
+
+    def get_parser(self, prog_name):
+        parser = super(Orchestrate, self).get_parser(prog_name)
+
+        parser.add_argument(
+            "cluster_config_file",
+            metavar="<cluster_config_file>",
+            help=_("File describing the cluster configuration"))
+        parser.add_argument(
+            "--cluster-id",
+            metavar="<cluster_id>",
+            help=_("OpenShift cluster ID"))
+        parser.add_argument(
+            "--infra-env-id",
+            metavar="<infra_env_id>",
+            help=_("OpenShift infrastruction environment ID"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        cluster_id = parsed_args.cluster_id
+        infra_env_id = parsed_args.infra_env_id
+
+        with open(parsed_args.cluster_config_file) as f:
```

Comment:
Do we want validate the config file? If some of the required values are missed then we can detect the errors at the beginning. Also, is it possible to add an example config file in a readme doc and state that which parameters are required and which are optional.

Reply:
Added the validation! The README does have an example config